src/backend/distributed/commands/alter_table.c

```diff
@@ -515,6 +519,8 @@ AlterTableSetAccessMethod(TableConversionParameters *params)
 TableConversionReturn *
 ConvertTable(TableConversionState *con)
 {
+	ErrorForUnsupportedCascadeObjects(con->relationId);
```

Comment:
Because we have this as our first check the error becomes misleading when the table is not distributed at all.

```
[local] nilsdijk@postgres:9700-24046=# SELECT undistribute_table('reference_table');
ERROR:  cannot undistribute table because the table is not distributed
Time: 0.855 ms

[local] nilsdijk@postgres:9700-24046=# ALTER EXTENSION plpgsql DROP VIEW owned_by_extension_view;
NOTICE:  Citus does not propagate adding/dropping member objects
HINT:  You can add/drop the member objects on the workers as well.
ALTER EXTENSION
Time: 6.489 ms

[local] nilsdijk@postgres:9700-24046=# SELECT undistribute_table('reference_table');
ERROR:  cannot undistribute table because the table is not distributed
Time: 0.359 ms
```

It would be better to first check if the table is actually distributed before doing more complex check like these. If it would cascade into an extension it is quite cumbersome to resolve the problem. Only to figure out it wasn't distributed in the first place.

Reply:
`IsCitusTable` check is already before `ConvertTable` function is called. The example you shared also shows the correct error, am I missing something?
Let me move this error check to the places `ConverTable` is called, where we check for other errors, so there is less confusion.

Reply:
Oh, wow, I missread the error, I thought it was the error you introduced. Let me double check if this was a copy past error, or that I was that blind